BackEnd/src/main/java/com/unosquare/admin_core/back_end/controller/DashboardController.java

```diff
@@ -36,22 +40,16 @@
 
     @GetMapping(value = "/getEmployeeEvents", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
-    public EmployeeEventViewModel getDashboardEventsByEmployeeId(@RequestParam(value = "date")  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)  LocalDate date) {
-            List<EventDTO> events = dashboardService.getEmployeeDashboardEvents(employeeCredentialsViewModel.getUserId(), date);
-            List<DashboardEventViewModel> results = events.stream().map(event -> modelMapper.map(event, DashboardEventViewModel.class)).collect(Collectors.toList());
-            EmployeeEventViewModel model = new EmployeeEventViewModel();
-            model.setEvents(results);
-            return model;
+    public List<EmployeeEventViewModel> getDashboardEventsByEmployeeId(@RequestParam(value = "date")  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)  LocalDate date) {
+        List<DashboardEventDTO> events = dashboardService.getEmployeeDashboardEvents(employeeCredentialsViewModel.getUserId(), date);
+        return events.stream().map(event -> modelMapper.map(event, EmployeeEventViewModel.class)).collect(Collectors.toList());
```

Comment:
Quite a lot going on in these controllers. Do we need to have the ModelMapper calls here? Is it not better to have that sort of stuff within the DashboardService?

For debugging purposes I'm not a huge fan of having a method call within another method call (ie: extract "employeeCredentialsViewModel.getUserId()").

Reply:
We map from Database entity <-> DTO <-> ViewModel. We use modelmapper to control the mapping. It should be present on the service and presentation layers.